tf5muxserver/mux_server.go

```diff
@@ -112,6 +114,8 @@ func NewMuxServer(ctx context.Context, servers ...func() tfprotov5.ProviderServe
 				result.resources[resourceType] = server
 				result.resourceSchemas[resourceType] = schema
 			}
+
+			result.resourceCapabilities[resourceType] = resp.ServerCapabilities
```

Comment:
So from what i'm reading, this is where the provider developers enabling of `PlanDestroy` would occur. Do you have an example of what that would look like? I was looking through the AWS provider for example and couldn't get a good idea of where they would enable this.

Also, is that a framework-only feature or does it only matter what version of Terraform core they are using?

Reply:
I believe that calling `PlanResourceChange` RPC during destroy was introduced in Terraform core in [v1.3.0](https://github.com/hashicorp/terraform/releases/tag/v1.3.0) for providers that have "opted in":

> provider plugin protocol: The Terraform CLI now calls PlanResourceChange for compatible providers when destroying resource instances.

[PlanDestroy has been set to true](https://github.com/hashicorp/terraform-plugin-framework/blob/main/internal/fwserver/server_getproviderschema.go#L30) in the Terraform Plugin Framework since [v0.10.0](https://github.com/hashicorp/terraform-plugin-framework/releases/tag/v0.10.0):

> tfsdk: The PlanResourceChange RPC on destroy is now enabled. To prevent unexpected Terraform errors, the framework attempts to catch errant provider logic in plan modifiers when destroying. Resource level plan modifiers may require updates to handle a completely null proposed new state (plan) and ensure it remains completely null on resource destruction. 

I think that this means that if a provider developer is using Terraform Plugin Framework >= `v0.10.0` with Terraform >= `v1.3.0` then `PlanResourceChange` RPC will be called during destroy.

Reply:
Ah okay, so it's not necessarily something a provider developer opts-into directly then. Something they opt-into by usage of the framework.

Thanks for the breakdown! 👍🏻

Reply:
Created TF-613 to walk through the background here. Sorry for the confusion.